datalad_next/iter_collections/annexworktree.py

```diff
@@ -0,0 +1,157 @@
+"""Report on the content of a Git-annex repository worktree
+"""
+from __future__ import annotations
+
+import logging
+from dataclasses import dataclass
+from os import linesep
+from pathlib import (
+    Path,
+    PurePath,
+    PurePosixPath,
+)
+from typing import (
+    Any,
+    Generator,
+)
+
+from more_itertools import intersperse
+
+from .gitworktree import (
+    GitWorktreeItem,
+    GitWorktreeFileSystemItem,
+    iter_gitworktree
+)
+from datalad_next.itertools import (
+    itemize,
+    load_json,
+    route_in,
+    route_out,
+    dont_process,
+)
+from datalad_next.runners import iter_subproc
+
+
+lgr = logging.getLogger('datalad.ext.next.iter_collections.annexworktree')
+
+
+linesep_bytes = linesep.encode()
```

Comment:
https://git-annex.branchable.com/news/version_10.20231129/

Git annex changes lineending behavior. I believe it would break this choice

Reply:
Thanks for the pointer. I changed the `itemize`-calls to use `sep=None`. This should take care of all known variants of line-endings.

Done in 28be76c2a96518899ae9fc4353ab04c4f0f12257

Reply:
I assume that means we pay an unconditional cost on non-windows platforms for that now. Maybe `sep=None` would have been better as a windows-only setting?

Reply:
This can be removed again. We declare the format.